src/MapUtil/MapUtil.js

```diff
@@ -304,11 +304,14 @@ export class MapUtil {
       Logger.error('Invalid input parameter for MapUtil.getLegendGraphicUrl.');
       return;
     }
+    const isTiledWMS = layer.getSource() instanceof OlSourceTileWMS;
```

Comment:
better

```js
const src = layer.getSource();
if(!src) {
  return undefined;
}
const isTiledWMS = src instanceof OlSourceTileWMS;
const isImageWMS = !isTiledWMS && src instanceof OlSourceImageWMS;
```

then the if below gets trivial and more cases can easily be added IMO